### PR TITLE
`Logos`: Adjust Capacity Planner to only load calibrated models on non-metal nodes

### PR DESCRIPTION
**Pull request by julia-rai**
## Summary by CodeRabbit

- **New Features**
  - Added support for uncapped speculative replicas across lanes and providers, while preventing duplicate or conflicting loads.
  - Manual model loading now validates model-specific calibration and capacity before scheduling.
  - Failed or stuck lanes remain blocked from repeated cold loads until they recover or disappear.

- **Bug Fixes**
  - Provider selection now favors calibrated, feasible workers.
  - Metal/MLX loading accepts only explicit override profiles and skips unsupported macOS VRAM checks.

- **Documentation**
  - Added guidance on calibration requirements and Metal/MLX exceptions.

### CODE REVIEW

File: logos/logos-orchestrator/src/logos/capacity/capacity_planner.py [hunk 1]
```diff
@@ -5446,6 +5471,37 @@ def _flush_deferred_kv_reconfigs(
...
+        """True when this profile must not be used to load a lane.
+
+        No profile at all always fails, even on Metal — only an override
+        profile is exempt there, not having nothing (see load_lane_manually).
+        """
+        if profile is None:
+            return True
+        if profile.residency_source in ("calibrated", "measured"):
+            return False
+        return not self._provider_is_metal(provider_id)
```
Comment (Claudia-Anthropica) [inline logos/logos-orchestrator/src/logos/capacity/capacity_planner.py]: @julia-rai [medium] The Metal exemption accepts every non-null profile rather than only an operator override. `seed_capabilities()` leaves a profile with `residency_source=None` in the runtime profile map even when that model is removed from advertised capabilities, so the manual-load picker can still select the database-linked model and this helper admits it; an HF-derived profile is admitted as well. Restrict the Metal exemption to `residency_source == "override"` while continuing to accept calibrated/measured profiles, and cover Metal profiles with `None` and `"hf"` sources.

<details><summary>🤖 Prompt for AI agents</summary>

In `logos/logos-orchestrator/src/logos/capacity/capacity_planner.py`, `_load_requires_calibration` exempts any non-null profile on Metal, allowing seeded or HF-derived uncalibrated records to load. Restrict the Metal exemption to profiles whose `residency_source` is `"override"`, and test `None` and `"hf"` sources on Metal.

</details>

**Review by Claudia-Anthropica — CHANGES_REQUESTED** [review]
@julia-rai The earlier ranking and missing-profile defects are fixed, and the captured test, lint, and build checks succeeded. One remaining Metal edge case permits non-override profile records to bypass the calibration gate and dispatch an uncalibrated load.

**Comment by wasnertobias** [comment]
@copilot resolve the merge conflicts in this pull request
